dashboard/app/controllers/rubrics_controller.rb

```diff
@@ -65,12 +65,18 @@ def get_ai_evaluations
     return head :not_found unless student
     return head :forbidden unless can?(:manage, student)
 
-    learning_goals = LearningGoal.where(rubric_id: permitted_params[:id])
-    # Get the most recent AI evaluation for each learning goal
-    learning_goal_ai_evaluations =
-      LearningGoalAiEvaluation.where(user_id: permitted_params[:student_id], learning_goal_id: learning_goals.pluck(:id)).
-        group_by(&:learning_goal_id).
-        map {|_, eval_list| eval_list.max_by(&:updated_at)}
+    learning_goal_ids = LearningGoal.where(rubric_id: permitted_params[:id]).pluck(:id)
+
+    # Get the most recent learning goals based on the most recent graded rubric
+    learning_goal_ai_evaluations = LearningGoalAiEvaluation.where(
+      learning_goal_id: learning_goal_ids,
+      rubric_ai_evaluation_id: RubricAiEvaluation.where(
+        id: LearningGoalAiEvaluation.where(
+          learning_goal_id: learning_goal_ids,
+        ).order(updated_at: :desc).pick(:rubric_ai_evaluation_id),
+        user_id: permitted_params[:student_id],
+      )
+    )
```

Comment:
It is two queries since you cannot limit a subquery in MySQL. So (1) it pulls out the latest RubricAiEvaluation record based on the latest updated LearningGoalAiEvaluation record, and from that (2) it pulls out all the other LearningGoalAiEvaluation records.

Not bad.

```
(2.9ms)  SELECT `learning_goal_ai_evaluations`.`rubric_ai_evaluation_id` FROM
`learning_goal_ai_evaluations` WHERE `learning_goal_ai_evaluations`.`learning_goal_id`
IN (5, 43, 8, 14, 15, 22, 37, 40) ORDER BY `learning_goal_ai_evaluations`.`updated_at` DESC LIMIT 1

LearningGoalAiEvaluation Load [NO INDEX] (3.6ms)  SELECT `learning_goal_ai_evaluations`.* FROM
`learning_goal_ai_evaluations` WHERE `learning_goal_ai_evaluations`.`learning_goal_id` IN
(5, 43, 8, 14, 15, 22, 37, 40) AND `learning_goal_ai_evaluations`.`rubric_ai_evaluation_id` IN
(SELECT `rubric_ai_evaluations`.`id` FROM `rubric_ai_evaluations` WHERE `rubric_ai_evaluations`.`id` = 1 AND
`rubric_ai_evaluations`.`user_id` = 1)
```

As it seems to suggest here, I need to add an index on the aggregate `learning_goal_id` and and `rubric_ai_evaluation_id`.... so I will try to add that to this PR before opening it.

Reply:
The alternative is to add `rubric_id` to RubricAiEvaluation, which makes some sense. Then, here, I would just query `RubricAiEvaluation.where(rubric_id: rubric.id).order(updated_at: :desc).first.learning_goal_ai_evaluations` or something like that.

I probably want an index on `updated_at` on these.